code/mob/melee_attack_procs.dm

```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
```

Comment:
any reason syringes and darts don't share a common type ancestor? something like `/obj/item/implant/projectile/body_visible/dart` and `.../syringe`
that way you can check for both at the same time; just include a var in the implant for the pull-out string (`syringe` vs `dart` vs whatever)

also has the bonus of avoiding strange edge cases- as written, if I have a barbed dart and an unbarbed syringe, nobody will be able to pull the syringe out.

Reply:
No reason in particular, sounds like a better idea